src/scmdata/filters.py

```diff
@@ -157,59 +156,43 @@ def pattern_match(  # pylint: disable=too-many-arguments,too-many-locals
         else values
     )
 
-    _meta_col = meta_col.copy()
-    if has_nan:
-        _meta_col.loc[
-            [np.isnan(i) if not isinstance(i, str) else False for i in _meta_col]
-        ] = "nan"
-
-    handle_as_number = np.issubdtype(meta_col.dtype, np.number)
-
-    if not handle_as_number and "" in _values:
-        _values.append("nan")
-
     for s in _values:
-        if not handle_as_number and (isinstance(s, str) or np.isnan(s)):
-            _regexp = (
-                str(s)
-                .replace("|", "\\|")
-                .replace(".", r"\.")  # `.` has to be replaced before `*`
-                .replace("*", ".*")
-                .replace("+", r"\+")
-                .replace("(", r"\(")
-                .replace(")", r"\)")
-                .replace("$", "\\$")
-                .replace("^", "\\^")
-            ) + "$"
-            pattern = re.compile(_regexp if not regexp else str(s))
-            try:
-                subset = [m for m in _meta_col if pattern.match(m)]
-            except TypeError as e:
-                # if it's not the cryptic pandas message we expect, raise
-                msg = str(e)
-                if msg != "expected string or bytes-like object":
-                    raise e  # pragma: no cover # emergency valve
-
-                error_msg = (
-                    "String filtering cannot be performed on column '{}', which "
-                    "contains NaN's, unless `has_nan` is True".format(_meta_col.name)
-                )
-                raise TypeError(error_msg)
-
-            depth = (
-                True
-                if level is None
-                else find_depth(_meta_col, str(s), level, separator=separator)
-            )
-            matches |= _meta_col.isin(subset) & depth
+        if isinstance(s, str) and s == "":
+            s = np.nan
+        if isinstance(s, str):
+            if s == "*" and level is None:
```

Comment:
Is this what we want or is it going to cause issues depending on the value of `regexp`?

Reply:
I think it removes the ability to have `*` as a value. I'll  add a check for the value of regexp, which defaults to True